File: Java/memory.java
// Kelas Memory Turunan Kelas Hardware
class Memory extends Hardware
{
    // Atribut Private
    private String frequency;
    private String MemorySize;
    private String supportsCuda;
 
    // Constructor
    Memory() 
    {

    }

    // Prosedur Getter dan Setter
    public String getFrequency() {
        return frequency;
    }

    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    public String getMemorySize() {
        return MemorySize;
    }

    public void setMemorySize(String memorySize) {
        MemorySize = memorySize;
    }

    public String getSupportsCuda() {
        return supportsCuda;
    }

    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }
    
}

File: C++/memory.cpp
#include <string>
#include <iostream>
#include "hardware.cpp"

using namespace std;

// Kelas Memory Turunan Kelas Hardware
class Memory : public Hardware
{
    // Atribut Private
    private:
        string frequency;
        string memorySize;
        string supportsCuda;

    // Atribut Public
    public:
        // Constructor
        Memory()
        {
        
        }

        // Constructor Input
        Memory(string frequency, string memorySize, string supportsCuda)
        {
            this->frequency = frequency;
            this->memorySize = memorySize;     
            this->supportsCuda = supportsCuda;
        }

        // Prosedur Setter
        void setFrequency(string frequency)
        {
            this->frequency = frequency;
        }

        void setMemorySize(string memorySize)
        {
            this->memorySize = memorySize;
        }

        void setSupportsCuda(string supportsCuda)
        {
            this->supportsCuda = supportsCuda;
        }

        // Fungsi Getter
        string getFrequency()
        {
            return this->frequency;
        }

        string getMemorySize()
        {
            return this->memorySize;
        }

        string getSupportsCuda()
        {
            return this->supportsCuda;
        }

        // Destructor
        ~Memory()
        {

        };
};